django/src/rdwatch/management/commands/ingest.py

```diff
@@ -0,0 +1,272 @@
+
+    bnd = rst.bands[0]
+    tilesize = 64
+
+    for xoffset in range(0, bnd.width, tilesize):
+        for yoffset in range(0, bnd.height, tilesize):
+            size = (
+                tilesize if xoffset + tilesize <= rst.width else rst.width - xoffset,
+                tilesize if yoffset + tilesize <= rst.height else rst.height - yoffset,
+            )
+            data = bnd.data(
+                offset=(xoffset, yoffset),
+                size=size,
+            )
+            yield GDALRaster(
+                {
+                    "width": tilesize,
+                    "height": tilesize,
+                    "scale": rst.scale,
+                    "origin": [
+                        rst.origin.x + xoffset * rst.scale.x,
+                        rst.origin.y + yoffset * rst.scale.y,
+                    ],
+                    "skew": rst.skew,
+                    "srid": rst.srid,
+                    "nr_of_bands": 1,
+                    "datatype": bnd.datatype(),
+                    "bands": [
+                        {
+                            "data": data,
+                            "size": size,
+                            "nodata_value": bnd.nodata_value,
+                        }
+                    ],
+                }
+            )
+
+
+def _ingest_geojson(
+    site_geojson_file: Path,
+    prediction_configuration: PredictionConfiguration,
+    tracking_configuration: TrackingConfiguration,
+) -> list[Site]:
+    """
+    Ingest the geojson file at the given path into the Site model,
+    returning a list of Site instances.
+    """
+    site_geojson = json.loads(site_geojson_file.read_text())
+
+    sites: list[Site] = []
+
+    # Ingest each site in the geojson
+    for i, site in enumerate(site_geojson["features"]):
+        properties: dict = site["properties"]
+
+        # The first item in the array, the ground truth, doesn't have an
+        # explicit label.
+        if i == 0:
+            assert "current_phase" not in properties
+            label = Site.GROUND_TRUTH
+        else:
+            label = label_mapping[properties["current_phase"]]
+
+        score = properties["score"]
+
+        geometry = GEOSGeometry(json.dumps(site["geometry"]))
+        assert isinstance(geometry, MultiPolygon | Polygon)
+
+        # Generate a MultiPolygon if the geometry is just a Polygon (this
+        # seems to be the case for all ground truth features).
+        if isinstance(geometry, Polygon):
+            geometry = MultiPolygon(geometry)
+
+        sensor_name = properties.get("sensor_name")
+
+        if not sensor_name:
+            assert label == Site.GROUND_TRUTH
+            saliency = None
+        else:
+            assert sensor_name
+
+            sensor_name = "L8" if sensor_name == "Landsat 8" else "S2"
+
+            satellite_timestamp = properties.get(
+                "observation_date", properties.get("start_date")
```

Comment:
This can be taken from `CropParse.timestamp` with the file name from each GeoJSON Feature's `properties.source`. It does include date and time information.